lib/main.js

```diff
@@ -1,33 +1,64 @@
 'use babel';
 
-// eslint-disable-next-line import/no-extraneous-dependencies, import/extensions
+// eslint-disable-next-line import/extensions, import/no-extraneous-dependencies
 import { CompositeDisposable } from 'atom';
-import Path from 'path';
-import * as helpers from 'atom-linter';
 
-// Local vars
-const regex = /(.+):(\d+)\t*(.+)/g;
+let helpers;
+let path;
 
-// Settings
-let executablePath;
-let rulesets;
+function loadDeps() {
+  if (!helpers) {
+    helpers = require('atom-linter');
+  }
+  if (!path) {
+    path = require('path');
+  }
+}
 
 export default {
   activate() {
-    require('atom-package-deps').install('linter-phpmd');
+    this.idleCallbacks = new Set();
+    let depsCallbackID;
+    const installLinterPhpmdDeps = () => {
+      this.idleCallbacks.delete(depsCallbackID);
+      if (!atom.inSpecMode()) {
+        require('atom-package-deps').install('linter-phpmd');
+      }
+      loadDeps();
+    };
+    depsCallbackID = window.requestIdleCallback(installLinterPhpmdDeps);
+    this.idleCallbacks.add(depsCallbackID);
 
     this.subscriptions = new CompositeDisposable();
+
     this.subscriptions.add(
       atom.config.observe('linter-phpmd.executablePath', (value) => {
-        executablePath = value;
+        this.executablePath = value;
+      }),
+      atom.config.observe('linter-phpmd.autoExecutableSearch', (value) => {
+        this.autoExecutableSearch = value;
+      }),
+      atom.config.observe('linter-phpmd.disableWhenNoConfigFile', (value) => {
+        this.disableWhenNoConfigFile = value;
       }),
-      atom.config.observe('linter-phpmd.rulesets', (value) => {
```

Comment:
Remember how I said to remove that check on the `executableTimeout` from `phpcs`? We actually need the same code, but for this setting, sorry!

Reply:
Hmm, working on multiple projects so I'll have to dig into that to see what you mean 😜

Reply:
[This code](https://github.com/AtomLinter/linter-phpmd/pull/115#discussion_r138484278).

Reply:
Ah yes, to make it backwards compatbile! 👍